File: internal/contracts/agentapi/commands.go
package agentapi

type CommandSensitivity string

const (
	CommandSensitivityStandard  CommandSensitivity = "standard"
	CommandSensitivitySensitive CommandSensitivity = "sensitive"
)

type CommandDefinition struct {
	ID          string
	Sensitivity CommandSensitivity
}

var knownCommandDefinitions = []CommandDefinition{
	{ID: "df_h", Sensitivity: CommandSensitivityStandard},
	{ID: "free_m", Sensitivity: CommandSensitivityStandard},
	{ID: "uptime", Sensitivity: CommandSensitivityStandard},
	{ID: "top_head", Sensitivity: CommandSensitivitySensitive},
	{ID: "journalctl_u", Sensitivity: CommandSensitivitySensitive},
	{ID: "systemctl_status", Sensitivity: CommandSensitivitySensitive},
	{ID: "dmesg_err", Sensitivity: CommandSensitivitySensitive},
	{ID: "docker_ps", Sensitivity: CommandSensitivitySensitive},
}

var knownCommandIDs = func() map[string]CommandDefinition {
	out := make(map[string]CommandDefinition, len(knownCommandDefinitions))
	for _, command := range knownCommandDefinitions {
		out[command.ID] = command
	}
	return out
}()

func IsKnownCommandID(commandID string) bool {
	_, ok := knownCommandIDs[commandID]
	return ok
}

func KnownCommandDefinitions() []CommandDefinition {
	return append([]CommandDefinition(nil), knownCommandDefinitions...)
}

func SensitivityForCommand(commandID string) (CommandSensitivity, bool) {
	command, ok := knownCommandIDs[commandID]
	if !ok {
		return "", false
	}
	return command.Sensitivity, true
}

func RequiresSensitiveConfirmation(commandID string) bool {
	sensitivity, ok := SensitivityForCommand(commandID)
	return ok && sensitivity == CommandSensitivitySensitive
}

File: web/src/config/commands.ts
export type CommandSensitivity = 'standard' | 'sensitive'

export type MonitoringInstanceCommand = {
  id: string
  name: string
  description: string
  sensitivity: CommandSensitivity
}

export const COMMAND_LIST: MonitoringInstanceCommand[] = [
  { id: 'df_h', name: 'df -h', description: '磁盘使用概览', sensitivity: 'standard' },
  { id: 'free_m', name: 'free -m', description: '内存使用情况', sensitivity: 'standard' },
  { id: 'uptime', name: 'uptime', description: '系统运行时间与负载', sensitivity: 'standard' },
  { id: 'top_head', name: 'top -bn1', description: '进程 CPU/内存排序快照', sensitivity: 'sensitive' },
  { id: 'journalctl_u', name: 'journalctl --lines=50', description: '系统日志最近 50 行', sensitivity: 'sensitive' },
  { id: 'systemctl_status', name: 'systemctl status', description: '所有 systemd unit 状态概览', sensitivity: 'sensitive' },
  { id: 'dmesg_err', name: 'dmesg --level=err', description: '内核错误日志', sensitivity: 'sensitive' },
  { id: 'docker_ps', name: 'docker ps', description: '运行中的 Docker 容器', sensitivity: 'sensitive' },
]

export const COMMAND_LABELS: Record<string, string> = Object.fromEntries(
  COMMAND_LIST.map((command) => [command.id, command.name]),
)

export const COMMAND_OPTIONS = [
  { value: '', label: '全部命令' },
  ...COMMAND_LIST.map((command) => ({ value: command.id, label: command.name })),
]

const COMMAND_SENSITIVITY: Record<string, CommandSensitivity> = Object.fromEntries(
  COMMAND_LIST.map((command) => [command.id, command.sensitivity]),
)

export function commandSensitivity(commandId: string): CommandSensitivity | undefined {
  return COMMAND_SENSITIVITY[commandId]
}
